pageserver/src/http/routes.rs

```diff
@@ -697,12 +698,45 @@ async fn tenant_create_handler(mut request: Request<Body>) -> Result<Response<Bo
     )
 }
 
-async fn tenant_config_handler(mut request: Request<Body>) -> Result<Response<Body>, ApiError> {
+async fn get_tenant_config_handler(request: Request<Body>) -> Result<Response<Body>, ApiError> {
+    let tenant_id: TenantId = parse_request_param(&request, "tenant_id")?;
+    check_permission(&request, Some(tenant_id))?;
+
+    let tenant = mgr::get_tenant(tenant_id, false)
+        .await
+        .map_err(ApiError::NotFound)?;
+
+    let state = get_state(&request);
+    let default_config = state.conf.default_tenant_conf;
+    let tenant_specific_config = tenant.tenant_specific_config();
+    let resulting_config = tenant_specific_config.merge(default_config);
+
+    let response = HashMap::from([
+        (
+            "tenant_specific_config",
+            serde_json::to_value(tenant_specific_config)
+                .context("serializing tenant config")
+                .map_err(ApiError::InternalServerError)?,
+        ),
+        (
+            "resulting_config",
```

Comment:
Naming NIT: I like `tenant_specific_overrides` and `effective_config` better.

Reply:
Renamed.